### PR TITLE
Settings provider

### PR DESCRIPTION
**Pull request by Brunettow**
Provider for Settings Preferences Implemented
Provider for Change Password Implemented

- Dummy token is used, will be changed after token part is implemented
- Actions due to response messages not handled

### CODE REVIEW

File: project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import 'dart:convert';
+import 'package:collaborative_science_platform/utils/constants.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+class SettingsProvider with ChangeNotifier {
+  String token = "cd018223f0564aa0f66af496d754c819494de87f"; //TEST
+  Future<void> changePassword(String oldPass, String newPass) async {
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart:8]: Here changePassword should take User user as argument.
changePassword(User user, String oldPass, String newPass)

File: project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    final Map<String, String> header = {
+      "Accept": "application/json",
+      "content-type": "application/json",
+      'Authorization': 'Token $token',
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart:12]: And here we should use user.token 

Before merging let's fix this. 

I know instead you used dummy token but still this is not a good usage. Instead setting user.token to something dummy is a better approach. 

For now don't add anything dummy, just add user.

File: project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          },
+        ),
+      );
+      print(response.statusCode);
+    } catch (e) {
+      rethrow;
+    }
+  }
+
+  Future<void> changePreferences(String bio, bool sendNotification, bool showActivity) async {
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart:32]: Same here. User user

File: project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    final Map<String, String> header = {
+      "Accept": "application/json",
+      "content-type": "application/json",
+      'Authorization': 'Token $token',
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/providers/settings_provider.dart:36]: user.token

File: project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/account_settings_form.dart [hunk 5]
```diff
@@ -77,6 +78,13 @@ class _AccountSettingsFormState extends State<AccountSettingsForm> {
     super.dispose();
   }
 
+  void changePreff() async {
+    
+      final settingsProvider = Provider.of<SettingsProvider>(context,listen: false);
+      await settingsProvider.changePreferences(aboutMeController.text, isSwitched, isSwitched2);
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/account_settings_form.dart:84]: Provider.of<Auth>(context, listen: false).user;

File: project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/change_password_form.dart [hunk 6]
```diff
@@ -15,23 +17,37 @@ class ChangePasswordForm extends StatefulWidget {
...
-    confirmPasswordFocusNode.dispose();
+    newPassController.dispose();
+    newPassFocusNode.dispose();
     super.dispose();
   }
 
+  void changePass() async {
+    
+      if (currentPass != "" && oldPassController.text != "") {
+        final settingsProvider = Provider.of<SettingsProvider>(context,listen: false);
```
Comment (omerfaunal) [inline project/FrontEnd/collaborative_science_platform/lib/screens/profile_page/widgets/change_password_form.dart:43]: Provider.of(context, listen: false).user;

**Review by omerfaunal — CHANGES_REQUESTED** [review]

**Comment by Brunettow** [comment]
Thank you for all the comments. I changed the code accordingly. I also added error and response messages for creating more a responsive page. There are also minor design changes.

**Review by omerfaunal — APPROVED** [review]

**Pull request merged**